File: hotel-app/src/main/java/eu/getsoftware/hotelico/service/booking/application/hotel/domain/infrastructure/dto/CustomerNotificationDTO.java
// path: hotelico/service/booking/application/hotel/domain/infrastructure/dto/CustomerNotificationDTO.java

package eu.getsoftware.hotelico.service.booking.application.hotel.domain.infrastructure.dto;

import eu.getsoftware.hotelico.clients.common.domain.ids.CustomerDomainEntityId;
import eu.getsoftware.hotelico.service.booking.adapter.out.persistence.hotel.model.InnerHotelEvent;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashMap;
import java.util.Map;

@Getter
@Setter
@NoArgsConstructor
public class CustomerNotificationDTO {

    private Long id = 0L;

    private String receiverId;

    private Long creationTime = 0L;

    /**
     * number of unread message from every sender
     */
    private Map<CustomerDomainEntityId, Integer> unreadChatProSenderCount;

    private Map<CustomerDomainEntityId, String> lastSeenOnlineHotelGuests;

    private String unreadChatsCounter;

    /**
     * current number of hotel activities
     */
    private Integer hotelUnreadActivitiesNumber;

    private Integer hotelGuestsNumber;

    private Integer myDealsNumber;

    private Integer myMenuOrdersNumber;

    private Integer[] hotelOnlineGuestIds;

    private Map<String, String> customerEvent = new HashMap<>();

    /**
     * last message from every sender
     */
    private Map<CustomerDomainEntityId, String> lastMessagesToMe;

    /**
     * last message time from every sender to customer
     */
    private Map<CustomerDomainEntityId, String> lastMessageTimesToMe;

    private Map<CustomerDomainEntityId, Boolean> lastMessageSeenByCustomer;

    private Map<CustomerDomainEntityId, Boolean> lastMessageDeliveredToCustomer;

    /**
     * current number of wall posts
     */
    private Integer wallMessageCounter;

    public void setCustomerEvent(Long customerId, Long hotelId, InnerHotelEvent event, String message, Long entityId) {
        customerEvent.put("senderDomainId", String.valueOf(customerId));
        customerEvent.put("hotelId", String.valueOf(hotelId));
        // customerEvent.put("event", event.getValue()); // if available
        customerEvent.put("message", message);
        // customerEvent.put("entity", event.getEntityString()); // if available
        customerEvent.put("entityId", String.valueOf(entityId));
    }

    public void setSocketPushCustomerEvent(String pushTitle, String pushMessage, String pushUrl, String pushIcon, String chatPartnerId) {
        customerEvent.put("pushTitle", pushTitle);
        customerEvent.put("pushIcon", pushIcon);
        customerEvent.put("pushUrl", pushUrl);
        customerEvent.put("pushMessage", pushMessage);
        customerEvent.put("chatPartnerId", chatPartnerId);
        customerEvent.put("pushTagId", "message" + chatPartnerId);
    }

    public void setMenuOrderInRoomNumber(int menuOrderInRoomNumber) {
        customerEvent.put("menuOrderInRoomNumber", String.valueOf(menuOrderInRoomNumber));
    }

    public void setLastMessageDelieveredToCustomer(Map<CustomerDomainEntityId, Boolean> lastMessageDelieveredToCustomer) {
        
    }
}

File: hotel-app/src/main/kotlin/eu/getsoftware/hotelico/hotelapp/application/hotel/domain/infrastructure/dto/CustomerNotificationDTO.kt
package eu.getsoftware.hotelico.hotelapp.application.hotel.domain.infrastructure.dto;

import eu.getsoftware.hotelico.hotelapp.adapter.out.hotel.model.HotelEvent
import java.util.Map

/**
 * Created by Eugen on 30.07.2015.
 */
data class CustomerNotificationDTO (var id: Long = 0)
{
    var receiverId: String? = null

    var creationTime: Long = 0

    /**
     * number of unread message from every sender
     */
    var unreadChatProSenderCount: Map<Long, Integer>? = null

    var lastSeenOnlineHotelGuests: Map<Long, String>? = null

    var unreadChatsCounter: String? = null

    /**
     * current number of hotel activities
     */
    //	private Integer hotelActivitiesNumber;

    var hotelUnreadActivitiesNumber: Int? = null

    var hotelGuestsNumber: Int? = null

    var myDealsNumber: Int? = null

    var myMenuOrdersNumber: Int? = null

    var hotelOnlineGuestIds: Array<Int>? = null

    var customerEvent: MutableMap<String, String>  = mutableMapOf()

//	private Integer hotelOfflineGuestsNumber;

    /**
     * new guests for today
     */
//	private String[] newHotelCustomers;

    /**
     * last message from every sender
     */
    var lastMessagesToMe: Map<Long, String>? = null

    /**
     * last message time from every sender to customer
     */
    var lastMessageTimesToMe: MutableMap<Long, String>? = null

    var lastMessageSeenByCustomer: MutableMap<Long, Boolean>? = null

    var lastMessageDelieveredToCustomer: MutableMap<Long, Boolean>? = null

    /**
     * currentNumber of wall posts
     */
    var wallMessageCounter: Int? = null

    fun setCustomerEvent(customerId: Long, hotelId: Long, event: HotelEvent, message: String, entityId: Long) {

        this.customerEvent["senderId"] = customerId.toString() 
        this.customerEvent["hotelId"] = hotelId.toString() 
        this.customerEvent["event"] = event.value
        this.customerEvent["message"] = message
        this.customerEvent["entity"] = event.entityString
        this.customerEvent["entityId"] = entityId.toString() 
    }

    fun setSocketPushCustomerEvent(pushTitle: String, pushMessage: String, pushUrl: String, pushIcon: String, chatPartnerId: String) {

        this.customerEvent["pushTitle"] = pushTitle
        this.customerEvent["pushIcon"] = pushIcon
        this.customerEvent["pushUrl"] = pushUrl
        this.customerEvent["pushMessage"] = pushMessage
        this.customerEvent["chatPartnerId"] = chatPartnerId
        this.customerEvent["pushTagId"] = "message$chatPartnerId"
    }

    fun setMenuOrderInRoomNumber(menuOrderInRoomNumber: Int) {
        this.customerEvent["menuOrderInRoomNumber"] = menuOrderInRoomNumber.toString()
    }
}
